livekit-agents/livekit/agents/llm/utils.py

```diff
@@ -102,49 +104,53 @@ def is_context_type(ty: type) -> bool:
     return is_call_context
 
 
-def build_legacy_openai_schema(
-    ai_function: AIFunction, *, internally_tagged: bool = False
-) -> dict[str, Any]:
-    """non-strict mode tool description
-    see https://serde.rs/enum-representations.html for the internally tagged representation"""
-    model = function_arguments_to_pydantic_model(ai_function)
-    info = get_function_info(ai_function)
-    schema = model.model_json_schema()
+def serialize_image(image: llm.ImageContent, cache_key: Any) -> dict[str, Any]:
+    if isinstance(image.image, str):
+        header, b64_data = image.image.split(",", 1)
+        encoded_data = base64.b64decode(b64_data)
+        media_type = header.split(";")[0].split(":")[1]
+        supported_types = {"image/jpeg", "image/png", "image/webp", "image/gif"}
+        if media_type not in supported_types:
+            raise ValueError(
+                f"Unsupported media type {media_type}. Must be jpeg, png, webp, or gif"
+            )
 
-    if internally_tagged:
         return {
-            "name": info.name,
-            "description": info.description or "",
-            "parameters": schema,
-            "type": "function",
+            "data_bytes": encoded_data,
+            "media_type": media_type,
+            "inference_detail": image.inference_detail,
         }
-    else:
+    elif isinstance(image.image, rtc.VideoFrame):
+        if cache_key not in image._cache:
+            opts = utils.images.EncodeOptions()
+            if image.inference_width and image.inference_height:
+                opts.resize_options = utils.images.ResizeOptions(
+                    width=image.inference_width,
+                    height=image.inference_height,
+                    strategy="scale_aspect_fit",
+                )
+            encoded_data = utils.images.encode(image.image, opts)
+            image._cache[cache_key] = encoded_data
+
         return {
-            "type": "function",
-            "function": {
-                "name": info.name,
-                "description": info.description or "",
-                "parameters": schema,
-            },
+            "data_bytes": image._cache[cache_key],
+            "media_type": "image/jpeg",
+            "inference_detail": image.inference_detail,
         }
+    raise ValueError("Unsupported image type")
 
 
-def build_strict_openai_schema(
-    ai_function: AIFunction,
-) -> dict[str, Any]:
-    """strict mode tool description"""
-    model = function_arguments_to_pydantic_model(ai_function)
-    info = get_function_info(ai_function)
-    schema = _strict.to_strict_json_schema(model)
-
+def serialize_fnc_item(fnc: llm.AIFunction, strict: bool = False) -> dict[str, Any]:
+    model = utils.function_arguments_to_pydantic_model(fnc)
+    if strict:
+        schema = _strict.to_strict_json_schema(model)
+    else:
+        schema = model.model_json_schema()
+    info = utils.get_function_info(fnc)
```

Comment:
This should normally be for OAI only (strict mode)

Reply:
Not fan of sharing this method. Let's just use the two functions accordingly on the plugins..